### PR TITLE
Fix `make_meta` while using categorical column with index

### LINKED ISSUE #9291: Exception while providing a categorical dictionary meta
There is an exception when forcing *name* to be an unknown category instead of a string object dtype.

```python
import dask
dd = dask.datasets.timeseries()
dd = dd.map_partitions(lambda part: part, meta={'id': 'int32', 'name': 'str', 'x': 'float64', 'y': 'float64'})      # Ok
dd = dd.map_partitions(lambda part: part, meta={'id': 'int32', 'name': 'category', 'x': 'float64', 'y': 'float64'}) # Raise
```
image

**Environment**:

- Dask version:  2022.7.0
- Python version: 3.9
- Operating System: Window 10
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by pavithraes**
- [x] Closes #9291
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/tests/test_utils_dataframe.py [hunk 1]
```diff
@@ -126,13 +126,19 @@ def __next__(self):
     assert len(meta.index) == 0
 
     # Categoricals
-    meta = make_meta({"a": "category"}, parent_meta=df)
+    meta = make_meta({"a": "category"}, parent_meta=df)  # fails
```
Comment (pavithraes) [inline dask/dataframe/tests/test_utils_dataframe.py:129]: @jsignell Thanks for [suggesting a solution](https://github.com/dask/dask/issues/9291#issuecomment-1202684720)!

Looks like the change fixes #9291 but breaks this line, with:
```
TypeError: Index(...) must be called with a collection of some kind, None was passed
```
Any idea what's happening?

Comment (jrbourbeau) [inline dask/dataframe/tests/test_utils_dataframe.py:129]: I think this is safe to remove

```suggestion
    meta = make_meta({"a": "category"}, parent_meta=df)
```

File: dask/dataframe/tests/test_utils_dataframe.py [hunk 2]
```diff
@@ -126,13 +126,19 @@ def __next__(self):
...
     assert meta.a.cat.categories[0] == UNKNOWN_CATEGORIES
     meta = make_meta(("a", "category"), parent_meta=df)
     assert len(meta.cat.categories) == 1
     assert meta.cat.categories[0] == UNKNOWN_CATEGORIES
 
+    # Categorials with Index
+    meta = make_meta({"a": "category", "b": "int64"}, index=idx)
+    assert len(meta.a.cat.categories) == 1
+    assert meta.index.dtype == "int64"
+    assert len(meta.index) == 0
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_utils_dataframe.py:140]: I think `Index` objects have an `.empty` attribute for these types of situations 

```suggestion
    assert not meta.index.empty
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @pavithraes!

File: dask/dataframe/tests/test_utils_dataframe.py (hunk 1 above)
Comment (pavithraes) [reply dask/dataframe/tests/test_utils_dataframe.py]: Oh yes, thanks for catching it!

**Review by jrbourbeau — APPROVED** [review]
Thanks @pavithraes!

**Pull request merged**